### PR TITLE
Tgates/DB-30042

### PR DESCRIPTION
**Pull request by thebithead**
Reconcile for K8s kind nuodbadmin spec.insightsEnabled.

Demo and test case is in the Jira ticket.

### CODE REVIEW

File: build/etc/charts/nuodbadmin-helm/templates/sts-admin.yaml [hunk 1]
```diff
@@ -75,7 +75,11 @@ spec:
         - { name: NUODB_DOMAIN_ENTRYPOINT,      value: "admin-0.domain" }
         - { name: NUODB_ALT_ADDRESS,            value: "$(POD_NAME).domain.$(NAMESPACE).svc" }
         - { name: NUODB_VARDIR,                 value: "/var/opt/nuodb/$(POD_NAME).$(NAMESPACE)" }
-        args: [ "nuoadmin" ]
+        args:
+          - "nuoadmin"
+          - "--"
+          - "-pendingReconnectTimeout=60000"
+          - "-processLivenessCheckSec=30"
```
Comment (vegichan) [inline build/etc/charts/nuodbadmin-helm/templates/sts-admin.yaml]: these are not your diffs, why do they show up here?

Comment (thebithead) [reply build/etc/charts/nuodbadmin-helm/templates/sts-admin.yaml]: I rebased from master.  I'm not sure why they are showing up here.

File: pkg/controller/nuodbadmin/nuodbadmin_ctl_internal.go [hunk 2]
```diff
@@ -168,50 +169,86 @@ func createNuodbAdminStatefulSet(thisClient client.Client, thisScheme *runtime.S
...
 		} else {
 			return configMap, reconcile.Result{Requeue: true, RequeueAfter: utils.ReconcileRequeueAfterDefault}, err
 		}
+	} else {
+		if isInsightsResource && !instance.Spec.InsightsEnabled {
+			log.Info("Hosted NuoDB Insights disabled - Deleting ConfigMap:",
+				"Namespace", request.Namespace, "Name", nuoResource.name)
+			err = thisClient.Delete(context.TODO(), configMap)
+			return nil, reconcile.Result{}, err
+		}
```
Comment (adriansuarez) [inline pkg/controller/nuodbadmin/nuodbadmin_ctl_internal.go]: All of the `reconcileX()` functions have identical structure. Is there a way to refactor this to remove code duplication?

Comment (thebithead) [reply pkg/controller/nuodbadmin/nuodbadmin_ctl_internal.go]: I'm not sure. There is no template or generic support in GoLang.

File: build/etc/charts/nuodbadmin-helm/templates/sts-admin.yaml (hunk 1 above)
Comment (adriansuarez) [reply build/etc/charts/nuodbadmin-helm/templates/sts-admin.yaml]: Are you sure you actually rebased rather than merged?

**Pull request merged**